gateway-util-launcher/src/test/java/org/apache/knox/gateway/launcher/ExtenderTest.java

```diff
@@ -0,0 +1,195 @@
+        properties.setProperty("class.path", "classpath");
+        Extender extender = new Extender(null, properties);
+
+        String configContent = this.getConfigContent("/new/classp/*,../classp");
+        extender.extractExtensionPathIntoProperty(configContent);
+
+        assertEquals("/new/classp/*,../classp;classpath", properties.getProperty("class.path"));
+    }
+
+    @Test
+    public void extractExtensionPathIntoPropertyEmptyTest() {
+        Properties properties = new Properties();
+        properties.setProperty("class.path", "classpath");
+        Extender extender = new Extender(null, properties);
+
+        String configContent = this.getConfigContent("");
+        extender.extractExtensionPathIntoProperty(configContent);
+
+        assertEquals("classpath", properties.getProperty("class.path"));
+    }
+
+    @Test
+    public void extractExtensionPathIntoPropertyEmptyWhitespaceTest() {
+        Properties properties = new Properties();
+        properties.setProperty("class.path", "classpath");
+        Extender extender = new Extender(null, properties);
+
+        String configContent = this.getConfigContent(" ");
+        extender.extractExtensionPathIntoProperty(configContent);
+
+        assertEquals("classpath", properties.getProperty("class.path"));
+    }
+
+    @Test
+    public void extractExtensionPathIntoPropertyNoConfigTest() {
+        Properties properties = new Properties();
+        properties.setProperty("class.path", "classpath");
+        Extender extender = new Extender(null, properties);
+
+        String configContent =
+                "<configuration>\n" +
+                        "    <property>\n" +
+                        "        <name>gateway.webshell.read.buffer.size</name>\n" +
+                        "        <value>1024</value>\n" +
+                        "        <description>Web Shell buffer size for reading</description>\n" +
+                        "    </property>\n" +
+                        "\n" +
+                        "    <!-- @since 2.0.0 websocket JWT validation configs -->\n" +
+                        "    <property>\n" +
+                        "        <name>gateway.websocket.JWT.validation.feature.enabled</name>\n" +
+                        "        <value>true</value>\n" +
+                        "        <description>Enable/Disable websocket JWT validation at websocket layer.</description>\n" +
+                        "    </property>\n" +
+                        "\n" +
+                        "    <!-- @since 1.5.0 homepage logout -->\n" +
+                        "    <property>\n" +
+                        "        <name>knox.homepage.logout.enabled</name>\n" +
+                        "        <value>true</value>\n" +
+                        "        <description>Enable/disable logout from the Knox Homepage.</description>\n" +
+                        "    </property>\n" +
+                        "</configuration>";
```

Comment:
You can add static XML content in `src/test/resources` and load them here to make the code more readable.

Reply:
I moved the XML to the gateway-site-test.xml file in the test/resources folder.